autoprotocol/util.py

```diff
@@ -179,8 +179,9 @@ def _check_container_type_with_shape(container_type, shape):
                 f"containers must have row and column counts either equal to "
                 f"1 or divisible by 2, but {shape} was specified."
             )
-
-    if shape["format"] == "SBS384" and container_wells < 384:
+ 
+# access reservoir attribute from predefined container type to determine if stampable
+    if shape["format"] == "SBS384" and container_wells < 384 and container_type.is_reservoir == False:
         raise ValueError(
             f"SBS384 transfers can only be executed in 384 well plates, but "
             f"container_type: {container_type} has {container_wells} wells."
```

Comment:
There may be cases where a user wants to define a reservoir without requiring the shape and container wells to have specific constraints. That said, it might be better to do this instead:

`if container_type.is_reservoir == True:
     pass
    elif if shape["format"] == "SBS384" and container_wells < 384:
            raise ValueError(
                f"SBS384 transfers can only be executed in 384 well plates, but "
                f"container_type: {container_type} has {container_wells} wells." `

Reply:
What about checking on a container's `capabilities` property.  One challenge with `is_reservoir`, is that there are many types of reservoirs which may not be 384 well stamping compatible, eg. columnwise 12 well reservoirs, creating a mismatch of how to accurately describe a container while distinguishing valid instructions.  Additionally, and a little outside the intended scope of this task, is the case of a 1536-well plate.  A 1536-well plate can be used with a 384-well liquid handling head, similarly to how a 384-well plate can with a 96-pipette head.  This `is_reservoir` ContainerType property would not apply and an error would be thrown. Again, outside the scope of the immediate task, but worth keeping in mind.

Consider something along the lines of:
```
elif shape["format"] == "SBS384" and "sbs384_compatible" not in container_type.capabilities:
...
```
This would allow setting the known compatibility on a per container basis, which is helpful to differentiate a subclass of reservoirs.

Reply:
Thanks yeah that makes sense. I submitted a pull request in a new branch and closed this one.